helix-core/src/test/java/org/apache/helix/manager/zk/TestZkHelixAdmin.java

```diff
@@ -560,6 +560,82 @@ public void testLegacyEnableDisablePartition() {
+
+    // Test the sanity check for resetPartition.
+    // resetPartition is expected to throw an exception when provided with a nonexist instance.
+    try {
+      admin.resetPartition(clusterName, "WrongTestInstance", testResource, Arrays.asList("1", "2"));
+      Assert.fail("Should throw HelixException");
+    } catch (HelixException expected) {
+      // This exception is expected because the instance name is made up.
+      Assert.assertEquals(expected.getMessage(), String.format(
+          "Can't reset state for %s.[1, 2] on WrongTestInstance, because WrongTestInstance"
+              + " has never existed in cluster %s", testResource, clusterName));
+    }
+
+    // resetPartition is expected to throw an exception when provided with a non-live instance.
+    try {
+      admin.resetPartition(clusterName, instanceName, testResource, Arrays.asList("1", "2"));
+      Assert.fail("Should throw HelixException");
+    } catch (HelixException expected) {
+      // This exception is expected because the instance is not alive.
+      Assert.assertEquals(expected.getMessage(), String.format(
+          "Can't reset state for %s.[1, 2] on %s, because %s does not live in cluster %s anymore",
+          testResource, instanceName, instanceName, clusterName));
+    }
+
+    HelixManager manager = initializeHelixManager(clusterName, instanceConfig.getInstanceName());
+    manager.connect();
+
+    // resetPartition is expected to throw an exception when provided with a non-exist resource.
+    try {
+      admin.resetPartition(clusterName, instanceName, testResource, Arrays.asList("1", "2"));
+      Assert.fail("Should throw HelixException");
+    } catch (HelixException expected) {
+      // This exception is expected because the resource is not added.
+      Assert.assertEquals(expected.getMessage(), String
+          .format("Can't reset state for %s.[1, 2] on %s, because %s is not added", testResource,
+              instanceName, testResource));
+    }
+
+    IdealState idealState = new IdealState(testResource);
+    idealState.setNumPartitions(3);
+    admin.addStateModelDef(clusterName, "MasterSlave", new MasterSlaveSMD());
+    idealState.setStateModelDefRef("MasterSlave");
+    idealState.setRebalanceMode(IdealState.RebalanceMode.FULL_AUTO);
+    admin.addResource(clusterName, testResource, idealState);
+
+    admin.enableResource(clusterName, testResource, true);
+    try {
+      admin.resetPartition(clusterName, instanceName, testResource, Arrays.asList("1", "2"));
+      Assert.fail("Should throw HelixException");
+    } catch (HelixException expected) {
+      // This exception is expected because partitions do not exist.
+      Assert.assertEquals(expected.getMessage(), String
+          .format("Can't reset state for %s.[1, 2] on %s, because not all [1, 2] exist",
+              testResource, instanceName));
+    }
+
+    // clean up
+    manager.disconnect();
+    admin.dropCluster(clusterName);
```

Comment:
Please add a TestHelper.verify() to verify that the cluster has indeed been fully deleted at this point.

Reply:
Could you add a verify() to make sure the cluster has been removed successfully?